appserver/website-bootstrap/explore.js

```diff
@@ -173,73 +177,159 @@ $(document).ready(function(){
-                    cityTrendUrl += '&to=' + endDateStr; //YYYY-mm-dd HH:MM:SS
-
-                    alert(cityTrendUrl);
-                    $.getJSON(cityTrendUrl, function(data){
-		                trends = data;
-	                }).then(function() {
-                        document.getElementById("city-time-title").innerHTML = "Top Trends for " + city + " Over the Specified Dates";
-                        let table = document.getElementById("display-timetrends-table");
-                        for(let i = 0; i < trends.length; i++){
-                            let row = table.insertRow(i+1);
-                            let cell1 = row.insertCell(0);
-                            //cell1.innerHTML =
-                            let cell2 = row.insertCell(1);
-                            //cell2.innerHTML =
-                            let cell3 = row.insertCell(2);
-                            //cell3.innerHTML =
-                            let cell4 = row.insertCell(3);
-                            //cell4.innerHTML =
-                            let cell5 = row.insertCell(4);
-                            //cell5.innerHTML =
-                            let cell6 = row.insertCell(5);
-                            //cell6.innerHTML =
-                        }
-                    });
-                }
+        if(city){ // check if a city was provided
+            if(woeid){ // check if the provided city was valid (spelled correctly and in our map)
+                document.getElementById("city-time-title").innerHTML = "Top Trends for " + city + " Over the Specified Dates"; // change the header of the table to match the city provided by the user
+                cityTrendUrl += ('woeid=' + woeid); // append the woeid to the url before handling the date inputs
+                handleDates(cityTrendUrl);
+            } else { // if the city provided by the user doesn't return a woeid from our map(spelled wrong or not a city we have data for), let the user know it is not a valid option
+                alert(city + " is not a valid option!")
             }
-        } else {
-            alert("Please provide a valid city name!");
+        } else { // if no city provided handle to date inputs without appending a woeid to the url
+            handleDates(cityTrendUrl);
         }
     });
 });
 
+function fillTemporalTable(temporalCityUrl){
+    let trends = null;
+    alert(temporalCityUrl);
+    $.getJSON(temporalCityUrl, function(data){
+        trends = data;
+    }).then(function() {
+        let table_body = document.createElement('tbody');
+        document.getElementById("display-timetrends-table").replaceChild(table_body, document.getElementById("display-timetrends-table-body"));
+        table_body.id = "display-timetrends-table-body";
+        if(trends.length > 50){
+            trends = trends.slice(0, 50);
+            populateCells(table_body, trends);
+        } else {
+            populateCells(table_body, trends);
```

Comment:
why not just have this?
 if(trends.length > 50){
            trends = trends.slice(0, 50);
}
populateCells(table_body, trends);

Reply:
fixed